src/reducers/plutus_script_by_hash.rs

```diff
@@ -0,0 +1,79 @@
+use pallas::crypto::hash::{Hash, Hasher};
+use pallas::ledger::primitives::alonzo;
+use pallas::ledger::primitives::alonzo::PlutusScript;
+use serde::Deserialize;
+
+use crate::{crosscut, model};
+
+#[derive(Deserialize)]
+pub struct Config {
+    pub key_prefix: Option<String>,
+    pub filter: Option<Vec<String>>,
+}
+
+pub struct Reducer {
+    config: Config,
+    address_hrp: String,
+}
+
+impl Reducer {
+    fn send_set_add(
+        &mut self,
+        slot: u64,
+        hash: Hash<28>,
+        cbor: PlutusScript,
+        output: &mut super::OutputPort,
+    ) -> Result<(), gasket::error::Error> {
+        let key = format!("{}", hash);
+        let member = hex::encode::<Vec<u8>>(cbor.into());
+
+        let crdt = model::CRDTCommand::LastWriteWins(key, member, slot);
+
+        output.send(gasket::messaging::Message::from(crdt))?;
+
+        Ok(())
+    }
+
+    fn reduce_alonzo_compatible_tx(
+        &mut self,
+        slot: u64,
+        tx: &alonzo::TransactionWitnessSet,
+        output: &mut super::OutputPort,
+    ) -> Result<(), gasket::error::Error> {
+        if let Some(plutus_scripts) = &tx.plutus_script {
```

Comment:
Your code is iterative but actually the way one should write this IMHO is that rather than returning OK at the end, one should rather map and collect all results and return them from this function. Now I am by far a Rust noob but perhaps Rust will take a first item out of all results or evaluate lazily and stop execution and return result on first error.

@scarmuega do you agree?

Reply:
Yes it return result on first error if we use the `?` operator